src/sme_lib/libs/sRand.cpp

```diff
@@ -0,0 +1,33 @@
+
+#include "libs/sRand.hxx"
+#include "Globals.hxx"
+#include "sms/rand.h"
+
+using namespace SME;
+
+TVec3f Util::Rand::randUnitVector()
+{
+    float x = (static_cast<float>(rand()) / static_cast<float>(0x7fff)) - 0.5;
+    float y = (static_cast<float>(rand()) / static_cast<float>(0x7fff)) - 0.5;
+    float z = (static_cast<float>(rand()) / static_cast<float>(0x7fff)) - 0.5;
+
+    float mag = sqrtf(x * x + y * y + z * z);
+    x /= mag;
+    y /= mag;
+    z /= mag;
+    return TVec3f(x, y, z);
```

Comment:
Here, I'd just inline the /= mag into the TVec3f initialization itself.

`return TVec3f(x / mag, y / mag, z / mag);`